test/unit/language-extensions/vararg.spec.ts

```diff
@@ -0,0 +1,38 @@
+import * as path from "path";
+import * as util from "../../util";
+import * as tstl from "../../../src";
+import { invalidVarargUse } from "../../../src/transformation/utils/diagnostics";
+
+const varargProjectOptions: tstl.CompilerOptions = {
+    types: [path.resolve(__dirname, "../../../language-extensions")],
+};
+
+test.each([
+    'const result = [...$vararg].join("")',
+    'let result: string; { result = [...$vararg].join(""); }',
+    'let result: string; if (true) { result = [...$vararg].join(""); }',
+    'let result: string; do { result = [...$vararg].join(""); } while (false);',
```

Comment:
What about a return statement? Think we should be able to test that.

Reply:
Is there a way to do a return statement in the global scope in ts?